File: java/GameAccountSimulator/src/gameAccountSimulator/LandFactory.java
package gameAccountSimulator;

public class LandFactory {

    public Land newLand(LandType type) {
        return createLand(type, 1);
    }

    public Land newLand(LandType type, int quantity) {
        return createLand(type, quantity);
    }
    
    private Land createLand(LandType type, int quantity) {
        Land land;

        switch (type) {
            case FARM:
                land = new Land("Farm", 50, quantity, 1);
                break;
            case LUMBER_MILL:
                land = new Land("Lumber Mill", 250, quantity, 5);
                break;
            case TAVERN:
                land = new Land("Tavern", 650, quantity, 12);
                break;
            case STONE_QUARRY:
                land = new Land("Stone Quarry", 2800, quantity, 50);
                break;
            case BARRAKS:
                land = new Land("Barraks", 9000, quantity, 150);
                break;
            case BLACKSMITH:
                land = new Land("Blacksmith", 14000, quantity, 250);
                break;
            case TEMPLE:
                land = new Land("Temple", 50000, quantity, 800);
                break;
            case VILLAGE:
                land = new Land("Village", 100000, quantity, 1400);
                break;
            case SHIPYARD:
                land = new Land("Shipyard", 180000, quantity, 2200);
                break;
            case CASTLE_KEEP:
                land = new Land("Castle Keep", 320000, quantity, 3200);
                break;
            case ROYAL_CASTLE:
                land = new Land("Royal Castle", 540000, quantity, 4500);
                break;
            case ROYAL_TRADE_ROUTE:
                land = new Land("Royal Trade Route", 1250000, quantity, 6200);
                break;
            case GOLD_MINE:
                land = new Land("Gold Mine", 4000000, quantity, 15000);
                break;
            default:
                land = null;
        }

        return land;
    }

}

File: objc/KingdomsSimulator/KingdomsSimulator/LandFactory.m
//
//  LandFactory.m
//  KingdomsSimulator
//
//  Created by Merxbauer Jaroslav on 02.10.11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "LandFactory.h"

@interface LandFactory()

- (Land*) createLandOfType: (LandType) landType withQuantity: (int) quantity;

@end

@implementation LandFactory

- (id)init
{
    self = [super init];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (Land*) newLandFromType:(LandType)type {
    return [self newLandFromType:type quantity:1];
}

- (Land*) newLandFromType:(LandType)type quantity:(int)quantity {
    return [self createLandOfType:type withQuantity:quantity];
}

- (Land*) createLandOfType:(LandType)landType withQuantity:(int)quantity {
    Land* land;
    
    switch (landType) {
        case FARM:
            land = [[Land alloc] initWithName:@"Farm" andStartingPrice:50 andQuantity:quantity andIncome:1];
            break;
        case LUMBER_MILL:
            land = [[Land alloc] initWithName:@"Lumber Mill" andStartingPrice:250 andQuantity:quantity andIncome:5];
            break;
        case TAVERN:
            land = [[Land alloc] initWithName:@"Tavern" andStartingPrice:650 andQuantity:quantity andIncome:12];
            break;
        case STONE_QUARRY:
            land = [[Land alloc] initWithName:@"Stone Quarry" andStartingPrice:2800 andQuantity:quantity andIncome:50];
            break;
        case BARRAKS:
            land = [[Land alloc] initWithName:@"Barraks" andStartingPrice:9000 andQuantity:quantity andIncome:150];
            break;
        case BLACKSMITH:
            land = [[Land alloc] initWithName:@"Blacksmith" andStartingPrice:14000 andQuantity:quantity andIncome:250];
            break;
        case TEMPLE:
            land = [[Land alloc] initWithName:@"Temple" andStartingPrice:50000 andQuantity:quantity andIncome:800];
            break;
        case VILLAGE:
            land = [[Land alloc] initWithName:@"Village" andStartingPrice:100000 andQuantity:quantity andIncome:1400];
            break;
        case SHIPYARD:
            land = [[Land alloc] initWithName:@"Shipyard" andStartingPrice:180000 andQuantity:quantity andIncome:2200];
            break;
        case CASTLE_KEEP:
            land = [[Land alloc] initWithName:@"Castle Keep" andStartingPrice:320000 andQuantity:quantity andIncome:3200];
            break;
        case ROYAL_CASTLE:
            land = [[Land alloc] initWithName:@"Royal Castle" andStartingPrice:540000 andQuantity:quantity andIncome:4500];
            break;
        case ROYAL_TRADE_ROUTE:
            land = [[Land alloc] initWithName:@"Royal Trade Route" andStartingPrice:1250000 andQuantity:quantity andIncome:6200];
            break;
        case GOLD_MINE:
            land = [[Land alloc] initWithName:@"Gold Mine" andStartingPrice:4000000 andQuantity:quantity andIncome:15000];
            break;
        default:
            land = nil;
    }
    
    return [land autorelease];
}

@end
